apps/qubit/modules/informationobject/actions/inventoryAction.class.php

```diff
@@ -0,0 +1,186 @@
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+class InformationObjectInventoryAction extends DefaultBrowseAction
+{
+  private static $levels;
+
+  public function execute($request)
+  {
+    $this->resource = $this->getRoute()->resource;
+
+    // Check that this isn't the root
+    if (!isset($this->resource->parent))
+    {
+      $this->forward404();
+    }
+
+    if (!self::showInventory($this->resource))
+    {
+      $this->forward404();
+    }
+
+    // Set title header
+    sfContext::getInstance()->getConfiguration()->loadHelpers(array('Qubit'));
+    $title = render_title($this->resource, false);
+    $this->response->setTitle("$title - Inventory list - {$this->response->getTitle()}");
+
+    if (empty($request->limit))
+    {
+      $request->limit = sfConfig::get('app_hits_per_page');
+    }
+
+    if ($this->getUser()->isAuthenticated())
+    {
+      $this->sortSetting = sfConfig::get('app_sort_browser_user');
+    }
+    else
+    {
+      $this->sortSetting = sfConfig::get('app_sort_browser_anonymous');
+    }
+
+    $resultSet = self::getResults($this->resource, $this->limit, $this->page, $this->sort);
+
+    // Page results
+    $this->pager = new QubitSearchPager($resultSet);
+    $this->pager->setPage($request->page ? $request->page : 1);
+    $this->pager->setMaxPerPage($request->limit);
+    $this->pager->init();
+  }
+
+  private static function getLevels()
+  {
+    if (self::$levels !== null)
+    {
+      return self::$levels;
+    }
+
+    $setting = QubitSetting::getByName('inventory_levels');
+    if (null === $setting || false === $value = unserialize($setting->getValue()))
+    {
+      return;
+    }
+
+    if (!is_array($value) || 0 === count($value))
+    {
+      return;
+    }
+
+    self::$levels = $value;
+
+    return $value;
+  }
+
+  private static function getResults(&$resource, &$limit = 10, &$page = 1, &$sort = 'identifierUp')
+  {
+    $query = new \Elastica\Query;
+    $query->setLimit($limit);
+    if (!empty($page))
+    {
+      $query->setFrom(($page - 1) * $limit);
+    }
+
+    $q = new \Elastica\Query\Bool;
+
+    $q1 = new \Elastica\Query\Term;
+    $q1->setTerm('ancestors', $resource->id);
+    $q->addMust($q1);
+    $q2 = new \Elastica\Query\Terms;
+    $q2->setTerms('levelOfDescriptionId', self::getLevels());
```

Comment:
What happens here if self::getLevel() returns null? (apparently /"return;" and "return null;" are equivalent in PHP)